devel/delete-from-ghcr

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+#
+# Delete Nextstrain images with a specific tag from GitHub Container registry.
+#
+set -euo pipefail
+
+if [[ $# -ne 1 ]]; then
+    echo "Please provide a tag." >&2
+    exit 1
+fi
+
+tag="$1"
+
+org=nextstrain
+
+for container in base base-builder
+do
+    echo "Finding version for $org/$container:$tag ..."
+
+    # Each tag should only correspond to one package version.
+    # Pushing an existing tag will untag the existing version and add the tag
+    # to the newly pushed version.
+    id=$(curl -s \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer $GITHUB_TOKEN" \
+          "https://api.github.com/orgs/$org/packages/container/$container/versions" \
+        | jq --arg TAG $tag '[ .[] | select(.metadata.container.tags[] == $TAG ) ] | .[0].id' \
+    )
+    if [[ $id = null ]]; then
+        echo "$org/$container:$tag was not found." >&2
+        exit 1
+    fi
+
+    echo "Version for $org/$container:$tag is $id."
+
+    # Don't use curl --fail since the error messages in the response are useful.
+    # Instead, parse error messages and exit accordingly.
+    echo "Deleting the package version for $org/$container:$tag ..."
+    deleteVersionError=$(curl -s \
+        -X DELETE \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer $GITHUB_TOKEN" \
+        "https://api.github.com/orgs/$org/packages/container/$container/versions/$id" \
+    )
+    echo "$deleteVersionError"
+
+    if [[ "$deleteVersionError" =~ "You cannot delete the last tagged version of a package. You must delete the package instead." ]]; then
```

Comment:
Did you look into using `skopeo delete` instead of GitHub's REST API? I would think that might be less overhead and GitHub Package's container registry would handle this sort of nuance for us. (But maybe not!)

Reply:
If `skopeo delete` won't work, you might also consider https://github.com/actions/github-script to run a little JS program where Octokit is automatically available and setup.

Reply:
Thanks, I hadn't thought about about either of those! Will look into it.

Reply:
`skopeo delete` [doesn't work on ghcr.io](https://github.com/nextstrain/docker-base/actions/runs/3373053143/jobs/5597277742). From [the docs](https://github.com/containers/skopeo/blob/62e698b5672236a3388afcf0a398fb0bb05f745b/docs/skopeo-delete.1.md):

> The effect of this is registry-specific; many registries don’t support this operation, or don’t allow it in some circumstances / configurations.

Reply:
`actions/github-script` is neat! Switched over to it in [force-push](https://github.com/nextstrain/docker-base/compare/b0e7395451b3cf316fec99a028e5b267a462a76e..2cb9afeafb9049319977d8793f955dfea8340ff5).

Reply:
:tada:

Reply:
I was kinda hoping Octokit had more sugar over the package tag to version id lookup and deletion of the last version stuff, but [alas it doesn't](https://octokit.github.io/rest.js/v19#packages-delete-package-version-for-org).